solutions/silver/baltic-12-mobile.mdx

```diff
@@ -100,3 +98,147 @@ int main() {
 	return 0;
 }
 ```
+
+---
+
+<Info title="Faster Solution">
+
+Although not necessary for full credit, there also exists an $\mathcal{O}(N)$ solution.
+
+Try to come up with it yourself before moving on!
+
+</Info>
+
+<Spoiler title="Faster Solution">
+
+The key observation is that not all stations contribute to the answer. If we ever have a station that isn't the closest to any point on the highway, we can simply ignore it.
+
+In fact, we can also ignore some points on the highway! Once we have this list of key stations, the number of points can be bounded by the number of key stations.
+
+<Spoiler title="How?">
+
+Let's say that we're looking for the point on the highway (excluding the endpoints) with the maximum distance to the nearest station, where there are only two stations, $a$ and $b$.
+
+Then, we can imagine a line perpendicular to the line segment formed by these two stations that passes through the midpoint. Everything on the left side is closer to $a$, and everything on the right side is closer to $b$.
+
+Thus, the maximum point is at the point where the perpendicular line intersects the highway! We only have to get the answer from these critical points, and since there are only two stations that could be closest to it, we can compute them in $\mathcal{O}(1)$.
+
+![Maximum Point](<baltic-12-mobile/perpendicular.png>)
+
+This intersection can be represented by $\frac{b_x^2 + b_y^2 - a_x^2 - a_y^2}{2(b_x - a_x)}$.
+
+From here on, we'll refer to this point as $p(a, b)$, where $a$ and $b$ are stations.
+
+</Spoiler>
+
+Now, let's consider cases where there exist unnecessary stations.
+
+Without loss of generality, let's assume all points are above the y-axis, and $a_x < b_x < c_x$.
```

Comment:
add smth like "if they aren't, we can just [do smth that would result in satisfaction w/o changing anything]"